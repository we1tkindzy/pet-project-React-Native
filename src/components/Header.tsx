import {View, StyleSheet, Button} from 'react-native';
import Themization from '../components/Themization';

interface HeaderProps {
  notMainPage?: boolean;
  navigation: any;
}

const Header = ({notMainPage, navigation}: HeaderProps) => {
  return (
    <View style={styles.settingsSection}>
      {notMainPage && (
        <Button
          color={'#371005'}
          title="Go back"
          onPress={() => navigation.goBack()}
        />
      )}

      <Themization />
    </View>
  );
};

export default Header;

const styles = StyleSheet.create({
  settingsSection: {
    width: '100%',
    backgroundColor: '#D9B597',
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    borderBottomLeftRadius: 15,
    borderBottomRightRadius: 15,
    padding: 10,
    marginBottom: 20,
  },
});
