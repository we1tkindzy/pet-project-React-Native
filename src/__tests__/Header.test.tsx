import {it} from '@jest/globals';
import renderer from 'react-test-renderer';
import Header from '../components/Header';

it('should renders Header component', () => {
  const header = renderer.create(<Header navigation={() => jest.fn()} />);
  expect(header).toMatchSnapshot();
});

it('should renders Header component with Back button', () => {
  const header = renderer.create(
    <Header notMainPage={true} navigation={() => jest.fn()} />,
  );
  expect(header).toMatchSnapshot();
});
