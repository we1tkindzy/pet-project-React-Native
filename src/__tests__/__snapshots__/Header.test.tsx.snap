// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renders Header component 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#D9B597",
      "borderBottomLeftRadius": 15,
      "borderBottomRightRadius": 15,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginBottom": 20,
      "padding": 10,
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "marginLeft": "auto",
        "width": "40%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#B6846B",
          "borderRadius": 25,
          "opacity": 1,
          "paddingHorizontal": 10,
          "paddingVertical": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "textAlign": "center",
            },
            {
              "color": "#ffffff",
            },
          ]
        }
      >
        dark theme
      </Text>
    </View>
  </View>
</View>
`;

exports[`should renders Header component with Back button 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#D9B597",
      "borderBottomLeftRadius": 15,
      "borderBottomRightRadius": 15,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginBottom": 20,
      "padding": 10,
      "width": "100%",
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
            {
              "color": "#371005",
            },
          ]
        }
      >
        Go back
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "marginLeft": "auto",
        "width": "40%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#B6846B",
          "borderRadius": 25,
          "opacity": 1,
          "paddingHorizontal": 10,
          "paddingVertical": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "textAlign": "center",
            },
            {
              "color": "#ffffff",
            },
          ]
        }
      >
        dark theme
      </Text>
    </View>
  </View>
</View>
`;
